### PR TITLE
COMMERCE-13016

### PR DESCRIPTION
**Pull request by brianikim**
Resent from here:

https://github.com/liferay-commerce/liferay-portal/pull/4563

https://liferay.atlassian.net/browse/COMMERCE-13016

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
@andrea-ale-sbarra ,

For: https://github.com/liferay-commerce/liferay-portal/pull/4563#discussion_r1452615323

@andrea-ale-sbarra , it is my fault for causing the confusion here. I accidentally did not commit the `data-qa-id` [implementation](https://github.com/liferay-commerce/liferay-portal/blob/c4bdeffd87d8e4206549669893fb7c6800f9436d/modules/apps/commerce/commerce-product-content-search-web/src/main/resources/META-INF/resources/specification_option_facets/view.jsp#L89C11-L89C11) on the specification facet that would make this work as intended with the assert on ordering as you would expect.

```
data-qa-id="<%= HtmlUtil.escapeAttribute(cpSpecificationOptionsSearchFacetDisplayContext.getCPSpecificationOptionTitle(facet.getFieldName())) + j %>">
```

I also updated the `expect` checks to be stricter. For my test case, the above data-qa-id lets me check the ordering because the ordering will be title+1, title+2, etc.

```
	await expect(
		_layoutsPage.page.getByTestId(optionCategory1.key + '1')
	).toBeVisible({timeout: 60 * 1000})

	await expect(
		_layoutsPage.page.getByTestId(optionCategory2.key + '2')
	).toBeVisible({timeout: 60 * 1000});
```

**Comment by brianikim** [comment]
ci:test:playwright

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:playwright

**Comment by brianikim** [comment]
ci:test:playwright

File: modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+			.filter({hasText: pageLabel})
+	).toBeVisible({timeout: 60 * 1000});
+
+	await expect(
+		_layoutsPage.page.getByTestId(optionCategory1.key + '1')
+	).toBeVisible({timeout: 60 * 1000});
+
+	await expect(
+		_layoutsPage.page.getByTestId(optionCategory2.key + '2')
+	).toBeVisible({timeout: 60 * 1000});
```
Comment (andrea-ale-sbarra) [inline modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts:172]: I still don't understand how the sorting is guaranteed here? I was expecting something like:
- find the list of specification by data-qa-id
- loop the entries of the the list
- verify each element to be in the right position

File: modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	const product2 = await _apiHelpers.headlessCommerceAdminCatalog.postProduct(
+		getProduct(catalog.id, 'Product2', specification2.key)
+	);
+
+	await _specificationFacetsPage.configureSearchOptions();
+
+	await _specificationFacetsPage.goToPage();
+
+	await expect(
+		_layoutsPage.page
```
Comment (andrea-ale-sbarra) [inline modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts]: Please move locator in *page file

File: modules/test/playwright/helpers/HeadlessCommerceAdminCatalogApiHelper.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	readonly basePath: string;
+
+	constructor(apiHelpers: ApiHelpers) {
+		this.apiHelpers = apiHelpers;
+		this.basePath = 'headless-commerce-admin-catalog/v1.0/';
+	}
+
+	async deleteCatalog(
+		apiHelpers = this.apiHelpers,
+		basePath = this.basePath,
```
Comment (andrea-ale-sbarra) [inline modules/test/playwright/helpers/HeadlessCommerceAdminCatalogApiHelper.ts:19]: this doesn't look right

Comment (andrea-ale-sbarra) [reply modules/test/playwright/helpers/HeadlessCommerceAdminCatalogApiHelper.ts:19]: why do you need to pass again these two parameters?

File: modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+const getCatalog = (catalogName: string) => {
+	const postCatalog = {
+		accountId: 0,
+		currencyCode: 'USD',
+		defaultLanguageId: 'en_US',
+		name: catalogName,
+	};
+
+	return postCatalog;
+};
```
Comment (andrea-ale-sbarra) [inline modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts:22]: Move this payload to the API helper

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Please also try to avoid the waits

File: modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts (hunk 2 above)
Comment (brianikim) [reply modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts]: Done as requested.

File: modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts (hunk 4 above)
Comment (brianikim) [reply modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts]: Done as requested.

File: modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts (hunk 1 above)
Comment (brianikim) [reply modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts]: Done as requested. 

```
	const panelList = await _specificationFacetsPage.panelList.all();

	const optionCategoryList = ['material', 'warranty'];

	for (let i = 0; i < optionCategoryList.length; i++) {
		await expect(panelList[i]).toHaveText(optionCategoryList[i]);
	}
```

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:playwright

File: modules/test/playwright/helpers/HeadlessCommerceAdminCatalogApiHelper.ts (hunk 3 above)
Comment (brianikim) [reply modules/test/playwright/helpers/HeadlessCommerceAdminCatalogApiHelper.ts]: You are correct. 
This was a necessary remnant from when the code was javascript and also utilizing Alessio's old commits. Removed as requested and fixed to correct type declarations.

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:playwright

**Comment by brianikim** [comment]
@andrea-ale-sbarra , I opted to combine the functional test commits into one again because I needed to refactor the tests due to:

https://github.com/brianchandotcom/liferay-portal/pull/146254

Thank you for being patient with me in this PR. Hopefully, everything is good to go. I believe I have fulfilled all of your requested changes in the latest iteration.

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:playwright

**Comment by brianikim** [comment]
@andrea-ale-sbarra , test may or may not fail due to https://github.com/brianchandotcom/liferay-portal/pull/146291.

That PR was merged while not respecting the changes in https://github.com/brianchandotcom/liferay-portal/pull/146283

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:playwright

**Comment by brianikim** [comment]
@andrea-ale-sbarra , test failure is caused by LPS-205933. I have notified the responsible team. See comment in the COMMERCE ticket for more details.

Please proceed with the code review to see if there are further changes needed. Again, thank you for being patient with me on this PR.

**Pull request closed without merging**